### ISSUE TITLE
Pytorch deprecation warning when running main.deepforest()

### ISSUE THREAD

**Issue by youngdjn** [weecology/DeepForest#369]
**Describe the bug**
When initiating a model with `m = main.deepforest()`, I get the following pytorch warning from the current torchvision version:

```
/home/exouser/anaconda3/lib/python3.9/site-packages/torchvision/models/_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
  warnings.warn(
/home/exouser/anaconda3/lib/python3.9/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=RetinaNet_ResNet50_FPN_Weights.COCO_V1`. You can also use `weights=RetinaNet_ResNet50_FPN_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)

```

**To Reproduce**
`from deepforest import main`
`m = main.deepforest()`

**Environment (please complete the following information):**
 - OS: Ubuntu 22
 - Python version and environment : Python 3.9

```
$ pip freeze | grep torch
pytorch-lightning==1.8.3.post1
torch==1.13.0
torchmetrics==0.11.0
torchvision==0.14.0

```

**Comment by Om-Doiphode** [comment]
I would like to work on this issue. Can you guide me on how to get started with it?

**Comment by ethanwhite** [comment]
Hi @Om-Doiphode! Welcome to the project. If you don't have a development version of DeepForest up and running already that's the starting point (if you need any help let us know) and then see if you can reproduce the warnings locally. After that I think the general approach would be:

1. Find the use or uses of `pretrained` in the code base that involve torchvision in someway (I think it's in `retinanet_resnet50_fpn`.
2. Look up the current docs for that function: https://pytorch.org/vision/master/models/generated/torchvision.models.detection.retinanet_resnet50_fpn.html
3. Based on those docs determine how to change the arguments to the function. The second part of the warning message even provides some help by telling you something about what change to make.

**Comment by Om-Doiphode** [comment]
Can you send me the instructions on how to build the development version?

**Comment by Om-Doiphode** [comment]
Is this the expected output? I have made the necessary changes in the model.py file 
[Solved](https://www.linkpicture.com/q/DeepForest.png)

**Comment by Om-Doiphode** [comment]
Is this the expected output? I have made the necessary changes in the model.py file
[Output](https://www.linkpicture.com/q/DeepForest.png)

**Comment by labeeb-7z** [comment]
This warning is caused due to torchvision upgrade from v0.12 to v0.13.
From v0.13  `pretrained` argument is deprecated and `weights` is introduced.

Also this warning is only shown when the model is loaded for the first time in your program.
![image](https://user-images.githubusercontent.com/94868003/217787667-f2941941-f7f3-4322-9822-d2006779a234.png)

![image](https://user-images.githubusercontent.com/94868003/217787615-2c3fb06a-3714-49f8-a3c1-1091642de2db.png)

The solution is just changing the `pretrained` argument to `weights` when the backbone model is loaded in `deepforest/model.py` as explained in the PyTorch docs you linked above.

I've made the change and tested by building locally, here's the output : 
![image](https://user-images.githubusercontent.com/94868003/217790584-cc4edc78-4802-439f-a9ba-ed6f53c86259.png)

 I've submitted a PR for this [here](https://github.com/weecology/DeepForest/pull/374)
